sandpack-client/src/clients/static/index.ts

```diff
@@ -81,17 +81,22 @@ export class SandpackStatic extends SandpackClient {
     }
   }
 
-  private injectDocType(content: FileContent): FileContent {
+  private validateHtml(content: FileContent): FileContent {
     // Make it a string
     let contentString = readBuffer(content);
 
-    // Add the DOCTYPE tag
-    const docTypeRegex = /<!DOCTYPE*>/gi;
-    if (!docTypeRegex.test(contentString)) {
-      contentString = `<!DOCTYPE html>\n${content}`;
-    }
+    // Check how long DOMParser takes
+    console.time('domparser');
+
+    const domParser = new DOMParser();
+    const doc = domParser.parseFromString(contentString, "text/html");
+    doc.documentElement.setAttribute("lang", "en");
```

Comment:
So, I definitely don't think we should be forcing all Sandpacks to use English as their HTML language, but it occurred to me that right now, there's no way to set the `lang` attribute in a Sandbox that doesn't render an `<html>` tag. This is kind of an issue; for example, the CSS [hyphens](https://developer.mozilla.org/en-US/docs/Web/CSS/hyphens) property only works when the `lang` attribute is set to "en" (and possibly other languages in the future).

Reply:
I think it's fine to default to `en`. Sandpack contains a lot of opinionated defaults, and we just need to provide a way to make it configurable, for example, persisting the html#lang value from the original document. Good insight, anyway!

Reply:
TIL, `DOMParser`! Very useful API!